### DISCUSSION TITLE
Running ASReview Simulation Mode in Google Colab

### DISCUSSION THREAD

**Discussion by jteijema** [asreview/asreview#595] [Show and tell]
As using your personal processing power seems very 2019, I wanted to see if running simulation mode works with google colab.

TL:DR, yes.

# Setting up
Before starting I set the runtime to GPU as this works well with the CUDA drivers for tensorflow. Keep in mind that there are some restrictions using the TPU and GPU.

![afbeelding](https://user-images.githubusercontent.com/28191416/117156466-01c4d380-adbe-11eb-8e78-8758a9884505.png)

**EDIT:** Setting to GPU is only useful for certain models! there seems to be no difference in simulation time for the NB model

# Installing
Installing ASReview and analysis packages is done with the following code: 

```
%%capture
!pip install asreview
!pip install asreview-statistics asreview-visualization
```

# Running
Running commands in colab is done using the ! prefix.

`!asreview simulate benchmark:van_de_Schoot_2017`

After having attached a google drive, this location can be used to store the h5 file for later use, without having to download this (probably big) file.

Pointing to the drive can be done like this: 

`--state_file drive/MyDrive/outputfile.h5`

The end result seems to be a lot quicker than doing it on my own machine.
![afbeelding](https://user-images.githubusercontent.com/28191416/117160953-d9d76f00-adc1-11eb-8cf6-938851f1c7f3.png)

# Plotting
Plotting is a small issue, as colab won't display matlibplot objects the right way. Luckily, the good people at ASReview created an API. Plotting a statefile would like like this: 

`from asreviewcontrib.visualization.plot import Plot`

```
with Plot.from_paths(["drive/MyDrive/Statefiles/PTSD_NB_TFIDF.h5"]) as plot:
	inc_plot = plot.new("inclusion")
	inc_plot.set_grid()
	inc_plot.set_xlim(0, 30)
	inc_plot.set_ylim(0, 101)
	inc_plot.set_legend()
	inc_plot.show()
	inc_plot.save("SOME_FILE.png")
```
![image](https://user-images.githubusercontent.com/28191416/120852097-0d9ce480-c57a-11eb-9250-978638f16b6c.png)

**Comment by J535D165** [comment]
This is super interesting. Never thought about using Colab for simulations. Maybe with a prefab notebook, we can make simulating the ASReview performance super easy for everyone. We can host them on our page and you can load them from Colab directly. Feel free to share your notebook (on the ASReview Github) with the public.

**Reply by jteijema** [reply]
Hi! This is the notebook I've been using so far: https://colab.research.google.com/drive/1V37uXyREtCcr7jDdm0whTAZR9D6WJzyY?usp=sharing

Would be amazing if we can provide this to people indeed.
